core/src/main/java/org/kiwix/kiwixmobile/core/utils/StorageUtils.kt

```diff
@@ -19,5 +19,5 @@ package org.kiwix.kiwixmobile.core.utils
 
 object StorageUtils {
   @JvmStatic fun getFileNameFromUrl(url: String?): String =
-    NetworkUtils.getFileNameFromUrl(url).replace(".meta4", "")
+    NetworkUtils.getFileNameFromUrl(url!!).replace(".meta4", "")
```

Comment:
it may be better to change the parameter to non null, this code basically guarantees an exception right now